src/main/java/com/training/assignments/Assignment4.java

```diff
@@ -0,0 +1,11 @@
+package com.training.assignments;
+
+public class Assignment4 {
+    public String getSubstring(String sentence, int startIndex, int endIndex) {
+        if (endIndex <= sentence.length() && startIndex < endIndex) {
+            return sentence.substring(startIndex, endIndex);
+        }
+        return "invalid data";
```

Comment:
Instead of returning a string here, lets throw an exception and see if we can unit test this exception?